src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -870,7 +873,19 @@ private void mapObservationUnitRows(
 		return allVariables;
 	}
 
-	private void addLocationIdVariable(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
+	void addLocationVariables(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
+		// check if LOCATION_ABBR already exists in the study, add if not present
+		final OptionalInt indexOfLocationAbbr = IntStream.range(0, environmentDetailAndConditionVariables.size())
```

Comment:
Does this change at the DatasetServiceImpl class to add LOCATION ABBR column (if not present) also affect XLS export? The scope of the ticket is just for CSV.

Reply:
XLS export use a different method to get the columns (getSubObservationSetVariables), also tested din po and the changes don't seem to affect xls export